### PR TITLE
Validate Apple Pay session on terminal selection

### PR DESCRIPTION
**Pull request by shahbaz-mehar**

### CODE REVIEW

**Review by BushraAsif — COMMENTED** [review]
Changes reviewed

File: assets/js/applepay.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+		"countryCode": applepay_ajax_obj.country,
+		"currencyCode": applepay_ajax_obj.currency,
+		"merchantCapabilities": [
+			"supports3DS"
+		],
+		"supportedNetworks": [
+			"visa",
+			"masterCard",
+			"amex",
+			"discover"
```
Comment (emicha) [inline assets/js/applepay.js:34]: I do not think we support discover

File: assets/js/applepay.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	const request = {
+		"countryCode": applepay_ajax_obj.country,
+		"currencyCode": applepay_ajax_obj.currency,
+		"merchantCapabilities": [
+			"supports3DS"
+		],
+		"supportedNetworks": [
+			"visa",
+			"masterCard",
+			"amex",
```
Comment (emicha) [inline assets/js/applepay.js:33]: You might want to have this as configuration in the plugin (in the UI), but with default `visa, masterCard, amex`

Comment (emicha) [reply assets/js/applepay.js:33]: Didn't we do that for Magento and NetSuite?

File: assets/js/applepay.js (hunk 1 above)
Comment (shahbaz-mehar) [reply assets/js/applepay.js]: @emicha I have removed discover.

File: assets/js/applepay.js (hunk 2 above)
Comment (shahbaz-mehar) [reply assets/js/applepay.js]: > You might want to have this as configuration in the plugin (in the UI), but with default `visa, masterCard, amex`

Yes, I have also created a configuration field for these on the plugin side and added selected`visa, masterCard, amex` options by default 
Screenshot 2023-03-17 at 3 37 29 PM

Comment (shahbaz-mehar) [reply assets/js/applepay.js]: > Didn't we do that for Magento and NetSuite?

No, we hardcoded the values there as well. But I will create tickets for each to get them dynamically.

**Review by oxeg — APPROVED** [review]

**Pull request merged**